src/main/java/uk/gov/digital/ho/hocs/casework/api/ActionDataExternalInterestService.java

```diff
@@ -0,0 +1,158 @@
+package uk.gov.digital.ho.hocs.casework.api;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.casework.api.dto.*;
+import uk.gov.digital.ho.hocs.casework.client.auditclient.AuditClient;
+import uk.gov.digital.ho.hocs.casework.client.auditclient.EventType;
+import uk.gov.digital.ho.hocs.casework.client.auditclient.dto.AuditPayload;
+import uk.gov.digital.ho.hocs.casework.client.infoclient.CaseTypeActionDto;
+import uk.gov.digital.ho.hocs.casework.client.infoclient.InfoClient;
+import uk.gov.digital.ho.hocs.casework.domain.exception.ApplicationExceptions;
+import uk.gov.digital.ho.hocs.casework.domain.model.*;
+import uk.gov.digital.ho.hocs.casework.domain.repository.ActionDataExternalInterestRepository;
+import uk.gov.digital.ho.hocs.casework.domain.repository.CaseDataRepository;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.*;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.ACTION_DATA_UPDATE_FAILURE;
+
+@Service
+@Slf4j
+public class ActionDataExternalInterestService implements ActionService {
+    private final ActionDataExternalInterestRepository actionDataExternalInterestRepository;
+    private final CaseDataRepository caseDataRepository;
+    private final InfoClient infoClient;
+    private final AuditClient auditClient;
+    private final CaseNoteService caseNoteService;
+
+    private static final String CREATE_CASE_NOTE_KEY = "RECORD_INTEREST";
+    private static final String UPDATE_CASE_NOTE_KEY = "UPDATE_INTEREST";
+
+    @Autowired
+    public ActionDataExternalInterestService(ActionDataExternalInterestRepository actionDataExternalInterestRepository,
+                                             CaseDataRepository caseDataRepository,
+                                             InfoClient infoClient,
+                                             AuditClient auditClient,
+                                             CaseNoteService caseNoteService) {
+        this.actionDataExternalInterestRepository = actionDataExternalInterestRepository;
+        this.caseDataRepository = caseDataRepository;
+        this.infoClient = infoClient;
+        this.auditClient = auditClient;
+        this.caseNoteService = caseNoteService;
+    }
+
+    @Override
+    public String getServiceDtoTypeKey() {
+        return ActionDataExternalInterestInboundDto.class.getSimpleName();
+    }
+
+    @Override
+    public String getServiceMapKey() {
+        return "RECORD_INTEREST";
+    }
+
+    @Override
+    public void create(UUID caseUuid, UUID stageUuid, ActionDataDto actionData) {
```

Comment:
You're missing validation of the case action type... e.g. if someone sent it through with any case action type, it would go through. You should really check that the type they are trying to save, is valid and active